pyomo/core/tests/unit/test_units.py

```diff
@@ -0,0 +1,446 @@
+from pyomo.core.expr import inequality
+import pyomo.core.expr.current as expr
+from pyomo.core.base.units_container import InconsistentUnitsError, UnitsError
+from six import StringIO
+
+try:
+    import pint
+    pint_available = True
+except ImportError:
+    pint_available = False
+
+def python_callback_function(arg1, arg2):
+    return 42.0
+
+@unittest.skipIf(pint_available is False, 'Testing units requires pint')
+class TestPyomoUnit(unittest.TestCase):
+
+    def test_PyomoUnit_NumericValueMethods(self):
+        m = ConcreteModel()
+        uc = units
+        kg = uc.kg
+
+        self.assertEqual(kg.getname(), 'kg')
+        self.assertEqual(kg.name, 'kg')
+        self.assertEqual(kg.local_name, 'kg')
+
+        m.kg = uc.kg
+
+        self.assertEqual(m.kg.name, 'kg')
+        self.assertEqual(m.kg.local_name, 'kg')
+
+        self.assertEqual(kg.is_constant(), False)
+        self.assertEqual(kg.is_fixed(), True)
+        self.assertEqual(kg.is_parameter_type(), False)
+        self.assertEqual(kg.is_variable_type(), False)
+        self.assertEqual(kg.is_potentially_variable(), False)
+        self.assertEqual(kg.is_named_expression_type(), False)
+        self.assertEqual(kg.is_expression_type(), False)
+        self.assertEqual(kg.is_component_type(), False)
+        self.assertEqual(kg.is_relational(), False)
+        self.assertEqual(kg.is_indexed(), False)
+        self.assertEqual(kg._compute_polynomial_degree(None), 0)
+
+        with self.assertRaises(TypeError):
+            x = float(kg)
+        with self.assertRaises(TypeError):
+            x = int(kg)
+
+        self.assertTrue(uc.check_units_consistency(kg < m.kg, uc))
+        self.assertTrue(uc.check_units_consistency(kg > m.kg, uc))
+        self.assertTrue(uc.check_units_consistency(kg <= m.kg, uc))
+        self.assertTrue(uc.check_units_consistency(kg >= m.kg, uc))
+        self.assertTrue(uc.check_units_consistency(kg == m.kg, uc))
+        self.assertTrue(uc.check_units_consistency(kg + m.kg, uc))
+        self.assertTrue(uc.check_units_consistency(kg - m.kg, uc))
+
+        with self.assertRaises(InconsistentUnitsError):
+            uc.check_units_consistency(kg + 3)
+
+        with self.assertRaises(InconsistentUnitsError):
+            uc.check_units_consistency(kg - 3)
+
+        with self.assertRaises(InconsistentUnitsError):
+            uc.check_units_consistency(3 + kg)
+
+        with self.assertRaises(InconsistentUnitsError):
+            uc.check_units_consistency(3 - kg)
+
+        # should not assert
+        # check __mul__
+        self.assertTrue(str(uc.get_units(kg*3)), 'kg')
+        # check __rmul__
+        self.assertTrue(str(uc.get_units(3*kg)), 'kg')
+        # check div / truediv
+        self.assertTrue(str(uc.get_units(kg/3.0)), 'kg')
+        # check rdiv / rtruediv
+        self.assertTrue(str(uc.get_units(3.0/kg)), '1 / kg)')
+        # check pow
+        self.assertTrue(str(uc.get_units(kg**2)), 'kg**2')
```

Comment:
Shouldn't this be `assertEqual`?  `assertTrue(x, 'kg')` onlt checks that `x.__bool__()` returns True and if it doesn't, then prints `'kg'` as the error message.

Reply:
Good catch - copy/paste bug - this was in several places.